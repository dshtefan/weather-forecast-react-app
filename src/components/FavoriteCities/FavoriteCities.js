import React from 'react';
import './FavoriteCities.scss';
import FavoriteCityContainer from "../../containers/FavoriteCityContainer";

const FavoriteCities = ({ cities }) => {
  return (
    <div className="city-list">
      {cities.map((el, i) => (<FavoriteCityContainer city={el} key={i} i={i}/>))}
    </div>
  )
};

export default FavoriteCities;