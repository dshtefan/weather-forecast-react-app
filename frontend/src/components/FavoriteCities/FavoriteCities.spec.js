import React from 'react';
import FavoriteCities from './';

describe('FavoriteCities', () => {
  const cities = [
    {
      city: 'Moscow',
      main: 'Rain',
      icon: '10',
      temp: '2',
      pressure: '997',
      humidity: '86',
      wind: '5',
      lat: '60',
      lon: '30'
    },
    {
      city: 'London',
      main: 'Rain',
      icon: '1',
      temp: '20',
      pressure: '907',
      humidity: '86',
      wind: '2',
      lat: '20',
      lon: '40'
    }
  ];

  describe('пустой список городов', () => {
    const favotiteCities = shallow(<FavoriteCities cities={[]} />);
    it('отображается корректно', () => {
      expect(shallowToJson(favotiteCities)).toMatchSnapshot();
    });
  });

  describe('список городов', () => {
    const favotiteCities = shallow(<FavoriteCities cities={cities} />);
    it('отображается корректно', () => {
      expect(shallowToJson(favotiteCities)).toMatchSnapshot();
    });
  });

});