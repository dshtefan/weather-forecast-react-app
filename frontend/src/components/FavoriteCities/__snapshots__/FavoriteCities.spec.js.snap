// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCity пустой список городов отображается корректно 1`] = `
<div
  className="city-list"
/>
`;

exports[`CurrentCity список городов отображается корректно 1`] = `
<div
  className="city-list"
>
  <Connect(CityListItem)
    city={
      Object {
        "city": "Moscow",
        "humidity": "86",
        "icon": "10",
        "lat": "60",
        "lon": "30",
        "main": "Rain",
        "pressure": "997",
        "temp": "2",
        "wind": "5",
      }
    }
    i={0}
    key="0"
  />
  <Connect(CityListItem)
    city={
      Object {
        "city": "London",
        "humidity": "86",
        "icon": "1",
        "lat": "20",
        "lon": "40",
        "main": "Rain",
        "pressure": "907",
        "temp": "20",
        "wind": "2",
      }
    }
    i={1}
    key="1"
  />
</div>
`;

exports[`FavoriteCities пустой список городов отображается корректно 1`] = `
<div
  className="city-list"
/>
`;

exports[`FavoriteCities список городов отображается корректно 1`] = `
<div
  className="city-list"
>
  <Connect(CityListItem)
    city={
      Object {
        "city": "Moscow",
        "humidity": "86",
        "icon": "10",
        "lat": "60",
        "lon": "30",
        "main": "Rain",
        "pressure": "997",
        "temp": "2",
        "wind": "5",
      }
    }
    i={0}
    key="0"
  />
  <Connect(CityListItem)
    city={
      Object {
        "city": "London",
        "humidity": "86",
        "icon": "1",
        "lat": "20",
        "lon": "40",
        "main": "Rain",
        "pressure": "907",
        "temp": "20",
        "wind": "2",
      }
    }
    i={1}
    key="1"
  />
</div>
`;
